src/Nethermind/Nethermind.Network.Stats/Model/InitiateDisconnectReason.cs

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+namespace Nethermind.Stats.Model;
+
+/// <summary>
+/// Nethermind level disconnect reason. Don't forget to add the corresponding Eth level disconnect reason in `InitiateDisconnectReasonExtension`.
+/// </summary>
+public enum InitiateDisconnectReason : byte
+{
+    IncomingConnectionRejectedTooManyPeer,
+    SessionAlreadyExist,
+    LostSessionDirectionDecision,
+    OppositeDirectionCleanup,
+
+    SnapServerNotImplemented,
+    IncompatibleP2PVersion,
+    InvalidCapability,
+    InvalidChainId,
+    InvalidGenesis,
+    ProtocolInitTimeout,
+    TxFlooding,
+    NoCapabilityMatched,
+
+    SyncPeerPoolBreachOfProtocol,
+    SyncPeerPoolUselessPeer,
+    SyncPeerPoolDropWorstPeer,
+    SyncPeerPoolRefreshFailed,
```

Comment:
Not sure if prefix here makes it more or less readable?

Reply:
Not sure too